core/src/utils/test/utils.ts

```diff
@@ -87,3 +89,45 @@ export const waitForFunctionTestContext = async (fn: any, params: any, interval
     }, interval);
   });
 };
+
+/**
+ * Given an element and optional selector, use the selector if
+ * it exists or get the node name of that element if not. Combine
+ * with the current mode to verify the correct classes exist.
+ *
+ * @param el: E2EElement - The element to verify classes on
+ * @param selector: string - A selector to use instead of the element tag name
+ *
+ * Examples:
+ * await checkComponentModeClasses(await page.find('ion-card-content'))
+ * => expect(el).toHaveClass(`card-content-{mode}`);
+ *
+ * await checkComponentModeClasses(await page.find('ion-card-content'), 'some-class')
+ * => expect(el).toHaveClass(`some-class-{mode}`);
+ */
+export async function checkComponentModeClasses(el: E2EElement, selector?: string) {
+  // If passed a selector to use, use that, else grab the nodeName
+  // of the element and remove the ion prefix to get the class selector
+  const component = selector !== undefined ? selector : el.nodeName.toLowerCase().replace('ion-', '');
+
+  const mode = await el.getProperty('mode');
+
+  // TODO remove
+  console.log(`component: ${component} classes: ${el.className}`);
+
+  expect(el).toHaveClass(`${component}-${mode}`);
+}
+
+/**
+ * Given an element, get the mode and verify it exists as a class
+ *
+ * @param el: E2EElement - the element to verify the mode class on
+ */
+export async function checkModeClasses(el: E2EElement) {
+  const mode = await el.getProperty('mode');
+
+  // TODO remove
+  console.log(`component: ${el.nodeName} classes: ${el.className}`);
```

Comment:
Don't forget to remove this

Reply:
I must've pushed the commit to remove this while you were reviewing somehow, it's gone now! 🙂